simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -661,6 +661,38 @@ class simd<std::int32_t, simd_abi::avx2_fixed_size<4>> {
       _mm_add_epi32(static_cast<__m128i>(lhs), static_cast<__m128i>(rhs)));
 }
 
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<std::int32_t, simd_abi::avx2_fixed_size<4>> operator>>(
+    simd<std::int32_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+    int rhs) noexcept {
+  return simd<std::int32_t, simd_abi::avx2_fixed_size<4>>(
+      _mm_srai_epi32(static_cast<__m128i>(lhs), rhs));
+}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<std::int32_t, simd_abi::avx2_fixed_size<4>> operator>>(
+    simd<std::int32_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+    simd<std::int32_t, simd_abi::avx2_fixed_size<4>> const& rhs) noexcept {
+  return simd<std::int32_t, simd_abi::avx2_fixed_size<4>>(
+      _mm_srav_epi32(static_cast<__m128i>(lhs), static_cast<__m128i>(rhs)));
+}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<std::int32_t, simd_abi::avx2_fixed_size<4>> operator<<(
+    simd<std::int32_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+    int rhs) noexcept {
```

Comment:
Please create an issue (or task) where we should replace int here (and elsewhere) with a private `simd_size_type`, just so we can in later comparisons with the standard have an easier time. Note `simd_size_type` is an implementation defined signed integral, so its natural for it to be int.

Reply:
I'll add a task to the issue #5674 so that we maintain a list of simd to-do's in one place instead of having them all scattered.

Reply:
yeah